File: Sitana.Framework/CrashReporter/Data/CrashReport.cs
using System;
using Sitana.Framework.DataTransfer;

namespace Sitana.Framework.CrashReporter.Data
{
	public class CrashReport: IEcsStructure
	{
		public string Crash;
		public DateAndTime Time;

        void IEcsStructure.Read(EcsReader reader)
		{
			Crash = reader.ReadString(0);
			Time = reader.ReadStructure<DateAndTime>(1);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			writer.Write(0, Crash);
			writer.Write(1, Time);
		}
	}
}

File: OtherLanguages/PHP/CrashReporter/Data/CrashReport.php
<?
class CrashReport
{
	public $crash;
	public $time;

	public function read($reader)
	{
		$this->crash = $reader->readString(0);
		$this->time = $reader->readStructure(1);
	}

	public function write($writer)
	{
		$writer->writeString(0, $this->crash);
		$writer->writeStructure(1, $this->time);
	}
}
?>